optuna/samplers/_tpe/sampler.py

```diff
@@ -590,3 +566,28 @@ def _get_observation_pairs(
             values[param_name].append(param_value)
 
     return values, scores
+
+
+def _split_observation_pairs(
+    config_vals: Dict[str, List[Optional[float]]], loss_vals: List[Tuple[float, float]], gamma: Callable[[int], int]
```

Comment:
Really just a suggestion but how about computing `n_below` at the caller? The interface might get simpler and probably easier to test.
```suggestion
    config_vals: Dict[str, List[Optional[float]]], loss_vals: List[Tuple[float, float]], n_below: int
```